README.md

```diff
@@ -1,126 +1,248 @@
+#### Parallelizable Tests
+This test suite consits of any test which completes in a short amount of time and does not require shared resources such as file descriptors, specific folders, or ports, and can therefore be run concurrently in different threads without side effects (hence, easily parallelized). These are mostly unit tests and [small tests](https://testing.googleblog.com/2010/12/test-sizes.html).
 
-# Also consider running the WASM spec tests for more coverage
-ctest -j $(nproc) -L wasm_spec_tests
+You can invoke them using `ctest` from a terminal in your Leap build directory.
+```bash
+ctest -j "$(nproc)" -LE _tests
 ```
 
-Some other tests are available and recommended but be aware they can be sensitive to other software running on the same host and they may **SIGKILL** other nodeos instances running on the host.
+#### WASM Spec Tests
+The WASM spec tests verify that our WASM execution engine is compliant with the web assembly standard. These are very [small](https://testing.googleblog.com/2010/12/test-sizes.html), very fast unit tests. However, there are over a thousand of them so the suite can take a little time to run. These tests are extremely CPU-intensive.
+
+You can invoke them using `ctest` from a terminal in your Leap build directory.
+```bash
+ctest -j "$(nproc)" -L wasm_spec_tests
 ```
-cd build
+We have observed severe performance issues when two virtual machines are running this test suite on the same physical host at the same time, for example in a CICD system. This can be resolved by disabling hyperthreading on the host.
 
-# These tests can't run in parallel but are recommended.
+#### Serial Tests
+The serial test suite consists of [medium](https://testing.googleblog.com/2010/12/test-sizes.html) component or integration tests that use specific paths, ports, rely on process names, or similar and cannot be run concurrently with other tests. Serial tests can be sensitive to other software running on the same host and they may `SIGKILL` other `nodeos` processes. These tests take a moderate amount of time to complete, but we recommend running them.
+
+You can invoke them using `ctest` from a terminal in your Leap build directory.
+```bash
 ctest -L "nonparallelizable_tests"
+```
 
-# These tests can't run in parallel. They also take a long time to run.
+#### Long-Running Tests
+The long-running tests are [medium-to-large](https://testing.googleblog.com/2010/12/test-sizes.html) integration tests that rely on shared resources and take a very long time to run.
+
+You can invoke them using `ctest` from a terminal in your Leap build directory.
+```bash
 ctest -L "long_running_tests"
 ```
+
+### Install
+Once you have [built](#build) Leap and [tested](#test) your build, you can install Leap to your system.
+
+We recommend installing the binary package you just built. Navigate to your Leap build directory in a terminal and run this command.
+```bash
+sudo apt-get update
+sudo apt-get install -y ./leap_*.deb
```

Comment:
Just a heads up that the filename of the binaries are different in 3.1 vs 3.2+ due to #120, what is used here is the 3.2 naming, but this PR is targeting 3.1.

Reply:
Should that change be back-ported to this branch? It kind of seems like a big deal that we aren't following Debian-family package naming conventions, if we intend to cut further releases from this branch.

If we don't intend to cut further releases from this branch, I probably shouldn't bother updating the `README.md`.

I am happy to change the commands here, just curious about your thoughts on that.

Reply:
I'm not actually so sure the 3.2 naming scheme is pedantically 'correct' either, due to the "ubuntu20.04" names stuck in it. That's not really what the revision field is for. Due to the uncertainty of this really being pedantically correct, I hadn't personally advocated for porting it to the 3.1.x branch. Especially because in 4.0 we expect to get rid of the "ubuntu20.04" in filenames thus truly making it correct at that point in time.

We might still make 3.1.3+ releases. It's still not entirely clear what our support lifecycle is: it's possible 3.1.x becomes an LTS of sorts should we go on to make a 3.3.x. Or not. 3.3x isn't a given anyways. But I think it's okay to continue to update the 3.1 readme.

Reply:
Fixed.